integration-tests/suites/duplicate_endpoints.go

```diff
@@ -61,27 +40,23 @@ func (s *DuplicateEndpointsTestSuite) killSocatProcess(port int) {
 func (s *DuplicateEndpointsTestSuite) SetupSuite() {
 
 	s.metrics = map[string]float64{}
-	s.executor = common.NewExecutor()
 	s.StartContainerStats()
-	s.collector = common.NewCollectorManager(s.executor, s.T().Name())
 
-	s.collector.Env["COLLECTOR_CONFIG"] = `{"logLevel":"debug","turnOffScrape":false,"scrapeInterval":20}`
-	s.collector.Env["ROX_PROCESSES_LISTENING_ON_PORT"] = "true"
+	collector := s.Collector()
 
-	err := s.collector.Setup()
-	s.Require().NoError(err)
+	collector.Env["COLLECTOR_CONFIG"] = `{"logLevel":"debug","turnOffScrape":false,"scrapeInterval":20}`
+	collector.Env["ROX_PROCESSES_LISTENING_ON_PORT"] = "true"
+
+	s.StartCollector(false)
 
-	err = s.collector.Launch()
-	s.Require().NoError(err)
 	time.Sleep(30 * time.Second)
 
 	processImage := config.Images().QaImageByKey("qa-socat")
 
 	containerID, err := s.launchContainer("socat", processImage, "TCP-LISTEN:80,fork", "STDOUT")
-
 	s.Require().NoError(err)
+
 	s.serverContainer = common.ContainerShortID(containerID)
-	s.waitForEndpoints()
```

Comment:
This can't be deleted. Read the comment for SetupSuite. The times of the scrapes have to be known and the times at which the listening endpoints are opened and closed have to be coordinated with the scrapes. Otherwise you will have flakes.

Reply:
The problem is the scrape timings are impossible to predict at this level. The current test does not strictly abide by the timings outlined in the comment, and `waitForEndpoints` does not account for the time required to download the database. I think this test needs a bit of a rethink, especially with the removal of the database.

Reply:
If the comments are not correct and you correct them? I think the line `s.Sensor().ExpectEndpointsN(s.T(), s.serverContainer, 60*time.Second, 1)` should give the same functionality as before.  This test has a longer scrape interval than the other tests to mitigate the problems that you outlined. The major problem that you outlined, is with a problem with the old method. What problems do you think there are with the test as it currently is?

Reply:
I think there has been a misunderstanding on my part on how this test works and the way in which collector reports endpoints. I'll update the docs and the general flow of this test to better reflect the behaviour under test.